### PR TITLE
Bugfix: target virt-launcher pod hangs when migration is cancelled

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
When virt-launcher starts, it searches for `qemu-kvm` PID.

If a virtual machine migration is created and then cancelled, before the virt-launcher detects the qemu-kvm process pid, the target virt-launcher is not cleaned up immediately and waits for the qemu-timeout (which is 3 minutes by default).

It will wait in the refresh monitor here https://github.com/kubevirt/kubevirt/blob/f77d50591ddd0f74c0c876e38fdf14ca3fe54be8/pkg/virt-launcher/monitor.go#L126.

This PR fixes this bug by stopping to search for the PID if the migration is cancelled.

This PR also adjusts a functional test that I confirm that fails without these changes and passes after them.

**Which issue(s) this PR fixes**:
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2161184

**How to reproduce**:
1. run `> ./cluster-up/kubectl.sh create -f examples/vmi-migratable.yaml`.
2. run `> ./cluster-up/kubectl.sh create -f examples/migration-job.yaml; sleep 5; ./cluster-up/kubectl.sh delete -f examples/migration-job.yaml`
3. See the target virt-launcher hangs for a few minutes.

**Release note**:

```release-note
Bugfix: target virt-launcher pod hangs when migration is cancelled.
```

### CODE REVIEW

File: tests/migration_test.go [hunk 1]
```diff
@@ -2715,7 +2715,7 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				Entry("[sig-storage][test_id:2228] with ContainerDisk and virtctl", newVirtualMachineInstanceWithFedoraContainerDisk, true),
 				Entry("[sig-storage][storage-req][test_id:2732] with RWX block disk and virtctl", decorators.StorageReq, newVirtualMachineInstanceWithFedoraRWXBlockDisk, true))
 
-			DescribeTable("Immediate migration cancellation after migration starts running", func(with_virtctl bool) {
+			FDescribeTable("Immediate migration cancellation after migration starts running", func(with_virtctl bool) {
```
Comment (prnaraya) [inline tests/migration_test.go:2718]:
```suggestion
			DescribeTable("Immediate migration cancellation after migration starts running", func(with_virtctl bool) {
```

Comment (prnaraya) [inline tests/migration_test.go:2718]: Left over?

```suggestion
			DescribeTable("Immediate migration cancellation after migration starts running", func(with_virtctl bool) {
```

Comment (iholder101) [reply tests/migration_test.go]: Thank you! Fixed!

p.s. we really need a linter to spot these out. FYI @nunnatsa :)

Comment (nunnatsa) [reply tests/migration_test.go]: @iholder101 /good-first-issue (;

**Review by jean-edouard — COMMENTED** [review]
Great catch! Just a few nits, might not be worth a new revision, up to you!
/approve

File: pkg/virt-launcher/monitor.go [hunk 2]
```diff
@@ -135,6 +137,9 @@ func (mon *monitor) refresh() {
 			} else if mon.gracePeriodStartTime != 0 {
 				log.Log.Infof("%s not found after shutdown initiated", mon.domainName)
 				mon.isDone = true
+			} else if cmdserver.ReceivedEarlyExitSignal() {
```
Comment (xpivarc) [inline pkg/virt-launcher/monitor.go:140]: Could we move this to the main loop?

File: tests/migration_test.go [hunk 3]
```diff
@@ -2724,25 +2724,51 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
...
+
+					if targetPodPhase != k8sv1.PodSucceeded && targetPodPhase != k8sv1.PodFailed {
+						return fmt.Errorf("pod phase is not expected to be %v", targetPodPhase)
+					}
+
+					return nil
+				}, 30*time.Second, 2*time.Second).ShouldNot(HaveOccurred(), "target migration pod is expected to disappear after migration cancellation")
+
+				By("Waiting for the migration object to disappear")
+				libwait.WaitForMigrationToDisappearWithTimeout(migration, 240)
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply tests/migration_test.go:2771]: +1

**Comment by iholder101** [comment]
Thanks for reviewing! Comments are addressed

@xpivarc PTAL

**Review by fossedihelm — COMMENTED** [review]
/lgtm
/hold
@xpivarc please unhold if everything is good for you
Thanks

**Comment by iholder101** [comment]
@xpivarc PTAL

**Comment by iholder101** [comment]
> @iholder101 Can you have a look on the flake lane? It could be related

After consulting with @brianmcarey, since the `pull-kubevirt-check-tests-for-flakes` lane might be flakie, I'll re-run the migration lane to make sure these tests don't fail again there.

/test pull-kubevirt-e2e-k8s-1.25-sig-compute-migrations

**Comment by brianmcarey** [comment]
> > @iholder101 Can you have a look on the flake lane? It could be related
> 
> After consulting with @brianmcarey, since the `pull-kubevirt-check-tests-for-flakes` lane might be flakie, I'll re-run the migration lane to make sure these tests don't fail again there.
> 
> /test pull-kubevirt-e2e-k8s-1.25-sig-compute-migrations

@xpivarc @iholder101 - it looks like the check-for-flakes lane is taking the latest provider (1.28 which is not yet gating) and it is just using a two node cluster where as the migration tests are usually run against a three node cluster so in this case it might not be fit for the migration tests. 

The compute-migrations lane has passed numerous times on this PR so I think we should be ok to rule out flakes here.

**Pull request merged**